### ISSUE TITLE
SSLPeerUnverifiedException

### ISSUE THREAD

**Issue by vRallev** [michel-kraemer/gradle-download-task#56]
Until today we used Gradle 2.13 and your download task plugin version 2.0.0 in our Android project. We wanted to upgrade Gradle to 2.14.1 and were forced to upgrade your plugin to latest version because of the ProgressLogger, too. 

I'm getting this exception now while downloading a file. It's very likely related to #48, but moving the download task to the root project isn't an option for us since the task depends on another task from the Android Gradle plugin. The exception isn't caused by Gradle or the Android plugin. If I don't change anything in the build script, use Gradle 2.13 and your plugin version 2.1.0, then everything is working. However, if I upgrade your plugin to 3.0.0 the exception occurs.

The only workaround I have so far is keep the old versions (living in the past isn't really an option) or to set `acceptAnyCertificate = true`, what I definitely don't want to do. Is there a different solution? What has changed in your plugin? 

```
Caused by: javax.net.ssl.SSLPeerUnverifiedException: Host name 'evernote.s3.amazonaws.com' does not match the certificate subject provided by the peer (CN=*.s3.amazonaws.com, O=Amazon.com Inc., L=Seattle, ST=Washington, C=US)
        at org.apache.http.conn.ssl.SSLConnectionSocketFactory.verifyHostname(SSLConnectionSocketFactory.java:465)
        at org.apache.http.conn.ssl.SSLConnectionSocketFactory.createLayeredSocket(SSLConnectionSocketFactory.java:395)
        at org.apache.http.conn.ssl.SSLConnectionSocketFactory.connectSocket(SSLConnectionSocketFactory.java:353)
        at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:134)
        at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:353)
        at org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:380)
        at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:236)
        at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:184)
        at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:88)
        at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110)
        at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:184)
        at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:71)
        at de.undercouch.gradle.tasks.download.DownloadAction.openConnection(DownloadAction.java:430)
        at de.undercouch.gradle.tasks.download.DownloadAction.execute(DownloadAction.java:191)
        at de.undercouch.gradle.tasks.download.DownloadAction.execute(DownloadAction.java:140)
        at de.undercouch.gradle.tasks.download.DownloadExtension.configure(DownloadExtension.java:44)
```

**Comment by michel-kraemer** [comment]
I don't think this issue has anything to do with #48. It seems to be related to a bug in the Apache HttpClient:
https://issues.apache.org/jira/browse/HTTPCLIENT-1636

I'll investigate this and get back to you.

**Comment by michel-kraemer** [comment]
Yes, it's related to the bug in the HttpClient. I was able to reproduce the issue. I then upgraded the HttpClient and everything worked as expected.

I'll fix this in master. The fix will be in the next version of gradle-download-task.

In the meantime you have two options/workarounds:

1) Set `acceptAnyCertificate = true` as you suggested
2) Include the plugin in your `build.gradle` as follows to update the dependency to the Apache HttpClient:

```
buildscript {
    repositories {
        jcenter()
    }
    dependencies {
        classpath 'de.undercouch:gradle-download-task:3.1.1'
        classpath 'org.apache.httpcomponents:httpclient:4.5.2'
    }
}
```

Cheers,
Michel

**Comment by vRallev** [comment]
Thanks for the quick fix and the temporary workaround. I prefer the second option much more!

**Comment by michel-kraemer** [comment]
You just have to remember to remove the dependency to the HttpClient when you update gradle-download-task the next time.

**Comment by vRallev** [comment]
Of course, that's how it looks like:
```groovy
    // remove http client when upgrading the download task plugin, that's a temporary workaround
    // see https://github.com/michel-kraemer/gradle-download-task/issues/56
    classpath 'de.undercouch:gradle-download-task:3.1.1'
    classpath 'org.apache.httpcomponents:httpclient:4.5.2'
```

**Comment by michel-kraemer** [comment]
I just released version 3.1.2. It should fix the issue.

Cheers,
Michel

**Comment by vRallev** [comment]
Yes, it works perfectly. Thanks @michel-kraemer!
